content/blog/getting-started-with-java-development-2023.asciidoc

```diff
@@ -0,0 +1,211 @@
+that also frees you from having to think about many aspects, which is great in particular when getting started.
+
+== 📝 Editor
+
+Java developers like to fight over their favourite build tools, and it's the same with editors and full-blown integrated development environments (IDEs).
+So whatever I'm going to say here, it's guaranteed a significant number of people won't like it ;)
+My suggestion is to use *VSCode* at the beginning.
+It's a rather light-weight editor, which comes with good Java developer support,
+e.g. for testing and debugging your code.
+It integrates with Maven-based projects very well and has a rich eco-system of plug-ins you can tap into.
+As your needs grow, you'll probably look for an IDE which come with even more advanced functionality, e.g. when it comes to refactoring your code.
+While I'm personally a happy user of the Eclipse IDE,
+most folks tend to use *IntelliJ* these days and it's thus what I'd recommend you too look into.
+It comes with a feature-rich free community edition which will help you a lot will the day-to-day tasks you'll encounter as a Java developer.
+Make sure to spend a few hours to learn the most important keyboard short-cuts,
+it will save you lots of time later on.
+
+== 🐳 Container Image
+
+In particular when you are going to work on an enterprise application,
+chances are that you'll deploy your application in form of a container image,
+so people can run your application on Docker or Kubernetes.
+Sticking to the recommendation on using Eclipse Temurin as your Java distribution,
+I'd suggest to use the *https://hub.docker.com/\_/eclipse-temurin/[Temurin image]* as the base for your application images,
+e.g. `eclipse-temurin:17` for Java 17.
+Just make sure to keep your image up to date,
+so you and your users benefit from updates to the base image.
+One base image you should avoid is the https://hub.docker.com/\_/openjdk[OpenJDK] one,
+which is officially deprecated and not recommended for production usage.
+
+== Application Framework
+
+While you could build a Java application solely based on the class library which comes with Java itself,
+depending on what it is you want to build,
+you'll be better off using an application framework which provides support for building web applications,
+accessing all kinds of databases, structuring your code via dependency injection and much more.
+Again, there's plenty of options in Java here,
+such as Spring Boot, Quarkus, Jakarta EE, Micronaut, Dropwizard, Helidon, and more.
+My personal recommendation here is to use *Quarkus*
+(it's the one I'm most familiar with, having worked for Red Hat before, who are the company behind this framework),
+or alternatively *Spring Boot*.
+Both are widely popular, integrate with a wide range of technologies (e.g. web frameworks and databases of all kinds),
+come with excellent developer tooling, and are backed by very active open-source communities.
+
+== Libraries
+
```

Comment:
Jackson for sure!

Reply:
Yepp, done. Added a handful. But also a disclaimer that the class library itself is very feature-rich and one should be conscious about adding dependencies.